src/basic/FStar.Errors.fs

```diff
@@ -833,31 +907,110 @@ let message_prefix =
      clear_prefix=clear_prefix;
      append_prefix=append_prefix}
 
-let findIndex l v = l |> List.index (function (e, _) when e=v -> true | _ -> false)
-let errno_of_error e = findIndex default_flags e
-
-let init_warn_error_flags = List.map snd default_flags
 
 let diag r msg =
-  if Options.debug_any() then add_one (mk_issue EInfo (Some r) msg None)
+  if Options.debug_any()
+  then add_one (mk_issue EInfo (Some r) msg None)
 
-let defensive_errno = errno_of_error Warning_Defensive
-let lookup flags errno =
-    if errno = defensive_errno && Options.defensive_fail ()
-    then CAlwaysError
-    else List.nth flags errno
+let warn_unsafe_options rng_opt msg =
+  match Options.report_assumes () with
+  | Some "warn" ->
+    add_one (mk_issue EWarning rng_opt ("Every use of this option triggers a warning: " ^msg) (Some 334))
```

Comment:
This shows for every DM4F effect, since we push `--admit_smt_queries` for typechecking the elaborated bind:
```
Opening file examples/dm4free/FStar.DM4F.Id.fst
Checking module: Id
Verifying implementation of FStar.DM4F.Id
(Warning 334) Every use of this option triggers a warning: admit_smt_queries
(Warning 334) Every use of this option triggers a warning: admit_smt_queries
```
(Also with no position)

Reply:
Ah, good catch.

Note, in case you write `fstar --report_assumes warn --admit_smt_queries true A.fst` then F* will report Warning 334 without any position information, since the problem is not in a source file, but in the command line arguments.

I suspect the use in DM4F is being triggered via the command line options somehow and so we miss an error location. I can fix that.

I also noticed that the warning shows up twice in this case (even in WarnOnUse.fst) ... not sure why. I think many such warnings are issued twice, because of two phase checking. I added some code to try to trap warn_on_use notifications so that they are only reported in phase1 or lax, but this seems to escape that check. 

But, we need a better overall strategy to ensure that the same error is not reported twice due to the phases. A brutal strategy here could be that when we report all issues, we simply do not report the same error on the same source position more than once. That would work, it would be easy to do, and it's foolproof: but it's just plastering over the problem of generating the same error twice in the first place.

Reply:
Ok, I implemented a patch in the error reporting code to only print unique errors.

That leaves the issue of what to do about the internal use of admit_smt_queries in DM4Free. The use of this admit is justified by meta arguments. And there's nothing the user can do to remove this assume (aside from not using DM4Free). So, what should we do? Suppress this warning?